streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -270,11 +270,12 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
         logInfo("Count = " + count + ", Running count = " + runningCount)
       }
       ssc.start()
-      ssc.awaitTerminationOrTimeout(500)
+      eventually(timeout(10.seconds), interval(10.millis)) {
+        assert(runningCount > 0)
+      }
       ssc.stop(stopSparkContext = false, stopGracefully = true)
       logInfo("Running count = " + runningCount)
       logInfo("TestReceiver.counter = " + TestReceiver.counter.get())
-      assert(runningCount > 0)
       assert(
         (TestReceiver.counter.get() == runningCount + 1) ||
           (TestReceiver.counter.get() == runningCount + 2),
```

Comment:
@tdas Could you clarify when `TestReceiver.counter.get() == runningCount + 2` is true?